csharp_dotnetcore/12.NLP-With-LUIS/LuisBot/AppInsights/MyAppInsightLuisRecognizer.cs

```diff
@@ -0,0 +1,124 @@
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MyAppInsightLuisRecognizer"/> class.
+        /// </summary>
+        /// <param name="application">The LUIS _application to use to recognize text.</param>
+        /// <param name="predictionOptions">The LUIS prediction options to use.</param>
+        /// <param name="includeApiResults">TRUE to include raw LUIS API response.</param>
+        public MyAppInsightLuisRecognizer(LuisApplication application, LuisPredictionOptions predictionOptions = null, bool includeApiResults = false, bool logOriginalMessage = false, bool logUserName = false)
+            : base(application, predictionOptions, includeApiResults)
+        {
+            LogOriginalMessage = logOriginalMessage;
+            LogUsername = logUserName;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether to log the <see cref="Activity"/> message text that came from the user.
+        /// </summary>
+        /// <value>If true, will log the Activity Message text into the AppInsight Custome Event for Luis intents.</value>
+        public bool LogOriginalMessage { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether to log the User name.
+        /// </summary>
+        /// <value>If true, will log the user name into the AppInsight Custom Event for Luis intents.</value>
+        public bool LogUsername { get; }
+
+        /// <summary>
+        /// Analyze the current message text and return results of the analysis (Suggested actions and intents).
+        /// </summary>
+        /// <param name="context">Context object containing information for a single turn of conversation with a user.</param>
+        /// <param name="ct">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <param name="logOriginalMessage">Determines if the original message is logged into Application Insights.  This is a privacy consideration.</param>
+        /// <returns>The LUIS results of the analysis of the current message text in the current turn's context activity.</returns>
+        public async Task<RecognizerResult> RecognizeAsync(ITurnContext context, CancellationToken ct, bool logOriginalMessage = false)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            // Call Luis Recognizer
+            var recognizerResult = await base.RecognizeAsync(context, ct);
+
+            var conversationId = context.Activity.Conversation.Id;
+
+            // Find the Telemetry Client
+            if (context.TurnState.TryGetValue(MyAppInsightsLoggerMiddleware.AppInsightsServiceKey, out var telemetryClient) && recognizerResult != null)
+            {
+                var topLuisIntent = recognizerResult.GetTopScoringIntent();
+                var intentScore = topLuisIntent.score.ToString("N2");
```

Comment:
I don't know what "N2" is. Consider adding a comment, as I don't think most folks will know this.

Reply:
Fixed (precision)